src/components/main-view/main-view.jsx

```diff
@@ -97,7 +98,7 @@ console.log(error);
           </Col>
           return movies.map(m => (
           <Col md={3} key={m._id}>
-          <MovieCard movie={m}  />
```

Comment:
This was the cause of the error that you were seeing about `undefined` `movieData`. According the `propTypes` the component `MovieCard` was a expecting a prop called `movieData=` instead of just `movie=`.